src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.InProcOrRemoteHostAnalyzerRunner.cs

```diff
@@ -215,7 +215,7 @@ internal class InProcOrRemoteHostAnalyzerRunner
         // TODO: filter in OOP https://github.com/dotnet/roslyn/issues/47859
         private static ImmutableDictionary<DocumentId, ImmutableArray<DiagnosticData>> Hydrate(ImmutableArray<(DocumentId documentId, ImmutableArray<DiagnosticData> diagnostics)> diagnosticByDocument, Project project)
             => diagnosticByDocument
-                .Where(entry => project.GetTextDocument(entry.documentId)?.SupportsDiagnostics() == true)
+                .Where(entry => project.GetTextDocument(entry.documentId)?.SupportsDiagnostics() != false)
```

Comment:
Does this need a comment or something? :smile:

Reply:
Yep, comment added